wandb/sdk/wandb_init.py

```diff
@@ -749,41 +737,34 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
             if result:
                 run_result = result.run_result
 
-            if not run_result:
-                error_message = "Error communicating with wandb process, exiting..."
-                error_message += f"\nFor more info see: {wburls.get('doc_start_err')}"
-                logger.error(
-                    "backend process timed out, exiting...\n"
-                    f"encountered error: {error_message}"
+            if run_result is None:
+                error_message = (
+                    f"Communication with the internal wandb process failed. Communication has timed out after {timeout} sec. "
+                    f"\nPlease refer to the documentation for additional information: {wburls.get('doc_start_err')}"
                 )
                 error = CommError(error_message)
```

Comment:
Add a `Comm`ent that it should rather be an InternalError, or subclass CommError into NetworkError & InternalError not to break legacy user code and provide separation for SDK and Gorilla errors?